numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -1299,6 +1304,96 @@ PyUFunc_MixedDivisionTypeResolver(PyUFuncObject *ufunc,
                                         type_tup, out_dtypes);
 }
 
+#define NOT_CONTIGUOUS(a) (PyArray_NDIM(a) >=2 && !( \
```

Comment:
Given we already check the correct strides in some of the cases below, I'd like to see this done correctly the first time, so we don't end up with a mix of overly strict and correct.

Reply:
It is complicated since a more optimized version should take the axis remapping into account, and the remapping info is not available in this function. It seems the only place we can really check for contiguous operands is in the inner loop.

Another question is if axis remapping breaks the current code. I will comment in the general PR comment section once I work out an opinion.

Reply:
I hadn't thought about the axis remapping coming into play here

>  It seems the only place we can really check for contiguous operands is in the inner loop.

I agree - is there a reason we can't do that?

Reply:
In dot we prefer blas even at the cost of copying. You are suggesting to reverse that and (at least for now) prefer to not use blas over copying?

Reply:
moving contigouous checks to inner loop, and adding a comment about the possible improvement